massa-execution-worker/src/interface_impl.rs

```diff
@@ -143,6 +143,29 @@ impl InterfaceClone for InterfaceImpl {
     }
 }
 
+/// Helper function that creates an amount from a NativeAmount
+fn amount_from_native_amount(amount: &NativeAmount) -> Result<Amount> {
+    let mantissa = amount
+        .mandatory_mantissa
+        .ok_or_else(|| anyhow!("Missing mantissa".to_string()))?;
+    let scale = amount
+        .mandatory_scale
+        .ok_or_else(|| anyhow!("Missing scale".to_string()))?;
+    let amount =
+        Amount::from_mantissa_scale(mantissa, scale).map_err(|err| anyhow!(format!("{}", err)))?;
+
+    Ok(amount)
+}
+
+/// Helper function that creates a NativeAmount from the amount internal representation
+fn amount_to_native_amount(amount: &Amount) -> Result<NativeAmount> {
```

Comment:
this shouldn't return a Result: it cannot fail

Reply:
done